hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -349,11 +349,19 @@ public void clear() {
     }
 
     @Override
-    protected Object invokeInternal(Data keyData, Data epData, Object[] arguments) {
+    protected Data invokeInternal(Data keyData, Data epData, Object[] arguments) {
+        Data valueData = null;
         try {
-            return super.invokeInternal(keyData, epData, arguments);
+            valueData = super.invokeInternal(keyData, epData, arguments);
```

Comment:
Seems it can be problematic to put result of entry-processor(EP) into near cache. Because EP can return any arbitrary type of object, result type is not limited to type of value. For example, a user defined type is a valid return type. Please see [this](https://github.com/jsr107/jsr107spec/blob/master/src/main/java/javax/cache/processor/EntryProcessor.java#L151). We can be on safe side by always invalidating near cache.